reccmp/isledecomp/compare/core.py

```diff
@@ -702,14 +712,17 @@ def _compare_function(self, match: MatchInfo) -> DiffReport:
         # Detect when the recomp function size would cause us to read
         # enough bytes from the original function that we cross into
         # the next annotated function.
+        assert match.orig_addr is not None
+        assert match.recomp_addr is not None
+        assert match.size is not None
```

Comment:
This is required a bunch of times. It might make sense to define another structure where all of these are not `None`, but that's out of the scope of this PR.